packages/smart-wallet/src/smartWallet.js

```diff
@@ -366,6 +428,87 @@ export const prepareSmartWallet = (baggage, shared) => {
             },
           });
         },
+
+        provideBrandToPurses() {
+          const brandToPurses = provideLazy(
+            walletPurses,
+            this.facets.helper,
+            _k => {
+              /** @type {MapStore<Brand, PurseRecord[]>} */
+              const store = makeScalarBigMapStore('purses by brand', {
+                durable: true,
+              });
+              return store;
+            },
+          );
+          return brandToPurses;
+        },
+
+        /**
+         * Provide a purse given a NameHub of issuers and their
+         * brands.
+         *
+         * We current support only one NameHub, agoricNames, and
+         * hence one purse per brand. But we store an array of them
+         * to facilitate a transition decentralized introductions.
+         *
+         * @param {Brand} brand
+         * @param {ERef<NameHub>} known - namehub with brand, issuer branches
+         * @returns {Promise<Purse | undefined>} undefined if brand is not well known
```

Comment:
the method call suggests the caller assumes the brand is well known. should this instead throw when that's not the case?

nit: strike "well" because "how known" is up to the caller

Reply:
further down I see that you are concerning this function with "well known" because it takes a remote call that you wouldn't want to have to make before calling this and within the function too.

I think it would help to clarify that in the name, like e.g. `providePurseIfKnownBrand`. Reading the current name again I see it could be interpreted that way "provide purse for a known brand and undefined otherwise" but the "otherwise" is not hinted at.

Reply:
yes, using `If` in the name is on target.